### ISSUE TITLE
Audit based policy not working with readonly rule

### ISSUE THREAD

**Issue by Ankurk99** [kubearmor/KubeArmor#989]
## Bug Report

If we have a audit based policy with "readonly" rule then we are not getting alert for the rule denial

**To Reproduce**
1) Apply the following policy
```
apiVersion: security.kubearmor.com/v1
kind: KubeArmorPolicy
metadata:
  name: wordpress-wordpress-4-8-apache-system-monitoring-write-under-dev-directory
  namespace: wordpress-mysql
spec:
  action: Audit
  file:
    matchDirectories:
    - dir: /dev/
      readOnly: true
      recursive: true
  message: Alert! File creation under /dev/ directory detected.
  selector:
    matchLabels:
      app: wordpress
  severity: 5
  tags:
  - NIST
  - NIST_800-53_AU-2
  - NIST_800-53_SI-4
  - MITRE
  - MITRE_T1036_masquerading
```
2) Exec into the wordpress pod and create a directory inside `/dev/` folder: `touch /dev/abc`

**Expected behavior**
The above action should have created an alert for illegal file creation as the `/dev/` path is readonly
Audit alerts should be visible for readonly rule

**Comment by 0xquark** [comment]
I would like to work on this issue :)

**Comment by nyrahul** [comment]
> I would like to work on this issue :)

hey @0xquark , thank you for the interest. Imo, the changes that would be required are primarily in the ebpf/telemetry handlers.
The aim is to check if the policy filespec matches in file path in ebpf open event and if the open event has "w/a" write/append flag enabled, then raise an audit event. Matching of filespec in ebpf open event works already but there is no check on the open syscall attributes currently and this needs to be handled.

CC: @achrefbensaad @daemon1024

**Comment by daemon1024** [comment]
A question with context to the issue. This is a audit policy, we raise alerts when the actual event happens and not in violation of that.

Isn't it fine that we didn't raise an alert because it wasn't a read only event?

We currently have this check for when open events are called with RDONLY they would raise an alert ig.
https://github.com/kubearmor/KubeArmor/blob/8406abab2b036f5e4da5d1c07f201309037fc1e8/KubeArmor/feeder/policyMatcher.go#L1013

**Comment by daemon1024** [comment]
Fixed the behaviour in #1626
